### PR TITLE
Add sam/sg81m layout

### PR DESCRIPTION
**Pull request by ryanbughuang**
## Description
Add 6.25u layout to sg81m 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sam/sg81m/info.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "height": 5.5,
     "layouts": {
         "LAYOUT": {
-            "layout": [
+            "layout_7u_space": [
```
Comment (drashna) [inline keyboards/sam/sg81m/info.json:9]:
```suggestion
        "LAYOUT_7u_space": {
            "layout": [
```

File: keyboards/sam/sg81m/info.json [hunk 2]
```diff
@@ -88,7 +88,92 @@
                 {"label":"K4C (F6,D6)", "x":15.25, "y":4.5},
                 {"label":"K59 (F7,D3)", "x":16.5, "y":4.25},
                 {"label":"K5A (F7,D4)", "x":17.5, "y":4.25}
+            ],
+            "layout_6.25u_space": [
```
Comment (drashna) [inline keyboards/sam/sg81m/info.json:92]:
```suggestion
            ],
        },
        "layout_6.25u_space": {
            "layout": [
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/sam/sg81m/info.json (hunk 1 above)
Comment (ryanbughuang) [reply keyboards/sam/sg81m/info.json:9]: thank you a lot for pointing that out!

**Review by zvecr — CHANGES_REQUESTED** [review]
This wont actually do anything as `keyboards/sam/sg81m/sg81m.h` has zero changes.

File: keyboards/sam/sg81m/sg81m.h [hunk 3]
```diff
@@ -33,5 +33,20 @@
     { K50,   K51,   K52,   K53,   K54,   K55,   K56,   K57,   K58,   K59,   K5A,   KC_NO, KC_NO, KC_NO }, \
 }
 
+#define LAYOUT_6.25u_space( \
```
Comment (drashna) [inline keyboards/sam/sg81m/sg81m.h:36]: That period will cause the compiler to fail. 
```suggestion
#define LAYOUT_6_25u_space( \
```

File: keyboards/sam/sg81m/info.json [hunk 4]
```diff
@@ -88,7 +88,93 @@
                 {"label":"K4C (F6,D6)", "x":15.25, "y":4.5},
                 {"label":"K59 (F7,D3)", "x":16.5, "y":4.25},
                 {"label":"K5A (F7,D4)", "x":17.5, "y":4.25}
+            ],
+        "LAYOUT_6.25u_space": {
```
Comment (drashna) [inline keyboards/sam/sg81m/info.json:92]:
```suggestion
        "LAYOUT_6_25u_space": {
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/sam/sg81m/sg81m.h (hunk 3 above)
Comment (ryanbughuang) [reply keyboards/sam/sg81m/sg81m.h]: thanks again!

**Review by drashna — APPROVED** [review]

File: keyboards/sam/sg81m/info.json [hunk 5]
```diff
@@ -88,7 +88,93 @@
                 {"label":"K4C (F6,D6)", "x":15.25, "y":4.5},
                 {"label":"K59 (F7,D3)", "x":16.5, "y":4.25},
                 {"label":"K5A (F7,D4)", "x":17.5, "y":4.25}
+            ],
+        "LAYOUT_6_25u_space": {
```
Comment (zvecr) [inline keyboards/sam/sg81m/info.json:92]:
```suggestion
            ],
        },
        "LAYOUT_6_25u_space": {
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/sam/sg81m/sg81m.h [hunk 6]
```diff
@@ -33,5 +33,20 @@
...
+    K40, K41, K42,                K45,           K47, K48, K49,   K4A, K4B, K4C,   K59, K5A  \
+) { \
+    { K00,   K01,   K02,   K03,   K04,   K05,   K06,   K07,   K08,   K09,   K0A,   K0B,   K0C,   K0D   }, \
+    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   K1D   }, \
+    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   K2C,   K2D   }, \
+    { K30,   K31,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B,   K3C,   K3D   }, \
+    { K40,   K41,   K42,   KC_NO, KC_NO, K45,   KC_NO, K47,   K48,   K49,   K4A,   K4B,   K4C,   K4D   }, \
+    { K50,   K51,   K52,   K53,   K54,   K55,   K56,   K57,   K58,   K59,   K5A,   KC_NO, KC_NO, KC_NO }, \
+}
+
```
Comment (zvecr) [inline keyboards/sam/sg81m/sg81m.h:50]: For backwards compatibility
```suggestion
#define LAYOUT LAYOUT_7u_space
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**